test/runnable/test22227.d

```diff
@@ -0,0 +1,16 @@
+// REQUIRED_ARGS: -debug -O -release
+// https://issues.dlang.org/show_bug.cgi?id=22277
+
+bool secret = false;
+
+void free(immutable void* x) pure nothrow
+{
+    debug secret = true;
+}
+
+void main()
+{
+    free(null);
+    if (!secret)
+        assert(0);
```

Comment:
Assert takes a bool. Why 0 instead of false ???

Reply:
That's the more common way to write it

Reply:
And what? We should use bad practices from C and ignore specs then?
Contemporary way is to use assert(false) https://github.com/dlang/dmd/blob/9d7d02589a6b8b503c084c1c5218cf9e09464410/test/compilable/noreturn_main.d#L19

Reply:
The spec uses `assert(0)` in its examples instead of `assert(false)`.